lib/db.js

```diff
@@ -118,6 +118,39 @@ class IndexTransaction {
         this.push({ type: 'del', key });
     }
 
+    /**
+     * Adds a condition for the transaction
+     *
+     * @argument {object} condition an object with the following attributes:
+     *                    {
+     *                      notExists: the object key
+     *                    }
+     *
+     * @throws {Error} an error described by the following properties
+     *                 - missingCondition if the condition is empty
+     *                 - pushOnCommittedTransaction if already committed
+     *                 - unsupportedConditionalOperation if the condition is not 'notExists'
+     *                 - missingKey if the key is missing from the op
+     *
+     * @returns {undefined}
+     */
+    addCondition(condition) {
+        if (this.closed) {
+            throw propError('pushOnCommittedTransaction',
+                            'can not add ops to already committed transaction');
+        }
+        if (condition === undefined || Object.keys(condition).length === 0) {
+            throw propError('missingCondition', 'missing condition for conditional put');
+        }
+        if (!('notExists' in condition)) {
```

Comment:
Looks like a better conditional check with `in` operator is, `if ('notExists' in condition === false) {}`, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in

Reply:
Why is it better?

Reply:
Also we might merge those two tests into a single `if (typeof condition.notExists !== 'string') { /*error*/ }` (for example)

Reply:
Thanks! your way much better and readable.
changed in [`35b801e` (#1545)](https://github.com/scality/Arsenal/pull/1545/commits/35b801e107f803d621aa54b80d46fca2b74a950a)

Reply:
better to read I guess